File: java/src/main/java/gov/doc/ntia/sigmf/ext/annotation/algorithm/DigitalFilterAnnotation.java
package gov.doc.ntia.sigmf.ext.annotation.algorithm;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import gov.doc.ntia.sigmf.Annotation;
import gov.doc.ntia.sigmf.serialization.DoubleSerializer;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class DigitalFilterAnnotation extends Annotation {

    private static final long serialVersionUID = 1L;

    @JsonProperty(value="ntia-algorithm:filter_type", required = false)
    protected String filterType;

    @JsonProperty(value="ntia-algorithm:FIR_coefficients", required = false)
    protected  Double[] firCoefficients;

    @JsonProperty(value="ntia-algorithm:IIR_numerator_coefficients", required = false)
    protected Double[] iirNumeratorCoefficients;

    @JsonProperty(value="ntia-algorithm:IIR_denominator_coefficients", required = false)
    protected Double[] iirDenominatorCoefficients;


    @JsonProperty(value="ntia-algorithm:frequency_cutoff", required = false)
    protected Double frequencyCutoff;


    @JsonProperty(value="ntia-algorithm:attenuation_cutoff", required = false)
    protected Double attenuationCutoff;


    @JsonProperty(value="ntia-algorithm:ripple_passband", required = false)
    protected Double ripplePassband;


    @JsonProperty(value="ntia-algorithm:attenuation_stopband", required = false)
    protected Double attenuationStopband;


    @JsonProperty(value="ntia-algorithm:frequency_stopband", required = false)
    protected Double frequencyStopband;

    public String getFilterType() {
        return filterType;
    }

    public void setFilterType(String filterType) {
        this.filterType = filterType;
    }

    public Double[] getFirCoefficients() {
        return firCoefficients;
    }

    public void setFirCoefficients(Double[] firCoefficients) {
        this.firCoefficients = firCoefficients;
    }

    public Double[] getIirNumeratorCoefficients() {
        return iirNumeratorCoefficients;
    }

    public void setIirNumeratorCoefficients(Double[] iirNumeratorCoefficients) {
        this.iirNumeratorCoefficients = iirNumeratorCoefficients;
    }

    public Double[] getIirDenominatorCoefficients() {
        return iirDenominatorCoefficients;
    }

    public void setIirDenominatorCoefficients(Double[] iirDenominatorCoefficients) {
        this.iirDenominatorCoefficients = iirDenominatorCoefficients;
    }

    public Double getFrequencyCutoff() {
        return frequencyCutoff;
    }

    public void setFrequencyCutoff(Double frequencyCutoff) {
        this.frequencyCutoff = frequencyCutoff;
    }

    public Double getAttenuationCutoff() {
        return attenuationCutoff;
    }

    public void setAttenuationCutoff(Double attenuationCutoff) {
        this.attenuationCutoff = attenuationCutoff;
    }

    public Double getRipplePassband() {
        return ripplePassband;
    }

    public void setRipplePassband(Double ripplePassband) {
        this.ripplePassband = ripplePassband;
    }

    public Double getAttenuationStopband() {
        return attenuationStopband;
    }

    public void setAttenuationStopband(Double attenuationStopband) {
        this.attenuationStopband = attenuationStopband;
    }

    public Double getFrequencyStopband() {
        return frequencyStopband;
    }

    public void setFrequencyStopband(Double frequencyStopband) {
        this.frequencyStopband = frequencyStopband;
    }



}

File: Csharp/sigmf/sigmf/ext/Annotationdir/algorithm/DigitalFilterAnnotation.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    public class DigitalFilterAnnotation : Annotation
    {
        [JsonProperty("algorithm:filter_type")]
        protected string filterType;

        [JsonProperty("ntia-algorithm:FIR_coefficients")]
        protected List<double> firCoefficients;

        [JsonProperty("ntia-algorithm:IIR_numerator_coefficients")]
        protected List<double> iirNumeratorCoefficients;

        [JsonProperty("ntia-algorithm:IIR_denominator_coefficients")]
        protected List<double> iirDenominatorCoefficients;

        [JsonProperty("ntia-algorithm:frequency_cutoff")]
        protected double frequencyCutoff;

        [JsonProperty("ntia-algorithm:attenuation_cutoff")]
        protected double attenuationCutoff;

        [JsonProperty("ntia-algorithm:ripple_passband")]
        protected double ripplePassband;

        [JsonProperty("ntia-algorithm:attenuation_stopband")]
        protected double attenuationStopband;

        [JsonProperty("ntia-algorithm:frequency_stopband")]
        protected double frequencyStopband;

        public string FilterType { get => filterType; set => filterType = value; }
        public List<double> FirCoefficients { get => firCoefficients; set => firCoefficients = value; }
        public List<double> IirNumeratorCoefficients { get => iirNumeratorCoefficients; set => iirNumeratorCoefficients = value; }
        public List<double> IirDenominatorCoefficients { get => iirDenominatorCoefficients; set => iirDenominatorCoefficients = value; }
        public double FrequencyCutoff { get => frequencyCutoff; set => frequencyCutoff = value; }
        public double AttenuationCutoff { get => attenuationCutoff; set => attenuationCutoff = value; }
        public double RipplePassband { get => ripplePassband; set => ripplePassband = value; }
        public double AttenuationStopband { get => attenuationStopband; set => attenuationStopband = value; }
        public double FrequencyStopband { get => frequencyStopband; set => frequencyStopband = value; }
    }
}
